extensions/ql-vscode/src/remote-queries/sarif-processor.ts

```diff
@@ -0,0 +1,137 @@
+import * as sarif from 'sarif';
+
+import { AnalysisAlert, ResultSeverity } from './shared/analysis-result';
+
+export class ProcessedSarif {
```

Comment:
Is there a reason to make this a class? Is it because you are planning on adding more behaviour to it over time? However, if these bits are going to need to be sent to the webview, or serialized to disk for query history, we should avoid classes with behaviour since that makes deserialization hard.

Reply:
It started as something but ended up being something else!

I've deleted the model completely.